static/js/main.js

```diff
@@ -0,0 +1,28 @@
+$(document).ready(function(){
+  if ($('.bar').length > 0) {
+    setTimeout(function start (){
+      
+      $('.bar').each(function(i){  
+        var $bar = $(this);
+        $(this).append('<span class="count"></span>')
+        setTimeout(function(){
+          $bar.css('width', $bar.attr('data-percent'));      
```

Comment:
This means the functionality doesn't work without Javascript. Using a CSS transform would be better.

Reply:
Though I've not been able to find a way to trigger a css transition when page loads. But while searching for the solution I came across common practices that have been suggested by various developers in such cases. One of it being to make the application fully functional without JS and use it only for ehnancements. I've followed the same. So what I've done now is to have a graph without transition when javascript is disabled, and a graph with transition otherwise.

Reply:
Yes, that's a good approach in general. Although you could use [CSS3 Animations](https://www.w3.org/TR/css3-animations/) as well - those can be triggered on page load.

Reply:
If we will use CSS3 animations, then we need to pass a variable (containing value of the width of the bar) from Django controller to CSS. There is no way to do so without writing CSS in templates. Will it be good to write CSS in HTML file? Should I switch from my previous approach to this one?

Reply:
Yeah that's a fair point. Maybe you could use a data attribute and `attr()`? Not sure if that works in a keyframe...

Anyway it's fine like this. It works without Javascript, which is the important part.